File: atcoder.jp/contests/abc167/tasks/abc167_b/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
  int A, B, C, K;
  cin >> A >> B >> C >> K;

  int a = min(A, K);
  int b = min(B, K - a);
  int c = K - a - b;
  int ans = a - c;

  cout << ans << endl;
}

File: atcoder.jp/contests/abc167/tasks/abc167_b/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];

const main = () => {
  const [A, B, C, K] = input().split(" ").map(Number);

  const a = Math.min(A, K);
  const b = Math.min(B, K - a);
  const c = K - a - b;
  const ans = a - c;

  console.log(ans);
};

main();
